src/CMakeLists.txt

```diff
@@ -40,16 +51,47 @@ if (NOT _ONEDPL_PSTL_OFFLOAD_BINARY_PATH)
     )
 
     add_custom_command(OUTPUT ${OUTPUT_VARS} COMMAND
-                ${CMAKE_COMMAND}
-                -DDPL_ROOT=${CMAKE_SOURCE_DIR}
-                -DVARS_TEMPLATE=${VARS_TEMPLATE}
-                -DOUTPUT_VARS=${OUTPUT_VARS}
-                -DPSTL_OFFLOAD_BINARY_PATH=${CMAKE_CURRENT_BINARY_DIR}
-                -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/generate_vars.cmake
-            )
-
+        ${CMAKE_COMMAND}
+        -DDPL_ROOT=${CMAKE_SOURCE_DIR}
+        -DVARS_TEMPLATE=${VARS_TEMPLATE}
+        -DOUTPUT_VARS=${OUTPUT_VARS}
+        -DPSTL_OFFLOAD_BINARY_PATH=${CMAKE_CURRENT_BINARY_DIR}
+        -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/generate_vars.cmake
+    )
     add_custom_target(set_env_vars DEPENDS ${OUTPUT_VARS})
     add_dependencies(pstloffload set_env_vars)
+
+    if (CMAKE_HOST_WIN32)
+        include(ExternalProject)
+        find_program(NMAKE_EXE NAMES nmake REQUIRED)
+        set(DETOURS_PATH ${PROJECT_BINARY_DIR}/src/project_detours-prefix/src/project_detours)
+        ExternalProject_Add(project_detours
+            GIT_REPOSITORY    https://github.com/microsoft/Detours.git
+            GIT_TAG           4b8c659f549b0ab21cf649377c7a84eb708f5e68
+            INSTALL_COMMAND   ""
+            CONFIGURE_COMMAND ""
+            # use CL to add additional flags to the Detours build
+            BUILD_COMMAND cd src && SET CL=/sdl
+            COMMAND ${NMAKE_EXE}
+            BUILD_IN_SOURCE on
+            STEP_TARGETS      build
+            BUILD_BYPRODUCTS ${DETOURS_PATH}/lib.X64/detours.lib
+        )
```

Comment:
Do we need to mention the use of detours in our documentation (cmake readme and/or pstl section of the guide)?

Are there any license implications? (I assume not since its being used as an external project, but I don't know)

Reply:
There are license implications, I believe. @ValentinaKats promised to add them in a separate commit.

Reply:
I added Detours as a third party program to the third-party-program.txt file directly to this PR, https://github.com/oneapi-src/oneDPL/pull/1359/commits/b09be05a855bd63f1d90f899625eda44764d5437.